### PR TITLE
[WIP]: Add unit tests for Bitcoin specific `strict_encoding`

### PR DESCRIPTION
**Pull request by rajarshimaitra**
This is a draft of unit tests for `strict_encoding` as per #60.

As of now the tests only makes an encode-decode round trip and verifies the integrity of the decoded object. 

Please suggest any form of optimization or missing test cases. 

Will remove [WIP] tag when seems ready for merge.

### CODE REVIEW

**Comment by rajarshimaitra** [comment]
Travis is failing at 
```
error: component 'rustfmt' for target 'x86_64-unknown-linux-gnu' is unavailable for download for channel nightly
Sometimes not all components are available in any given nightly.
The command "./contrib/test.sh" exited with 1.
```
Which is not occurring at my local and it seem like a rust issue discussed here https://github.com/rust-lang/rustfmt/issues/3404

**Comment by dr-orlovsky** [comment]
Yes, travis failure related to rust itself and it will be fixed only with the next nightly release

File: Cargo.toml [hunk 1]
```diff
@@ -32,7 +32,7 @@ amplify_derive = "~0.1.2"
 lazy_static = "~1.4.0"
 bech32 = "~0.7.2"
 ed25519-dalek = "~1.0.0-pre.3"
-bitcoin = { git = "https://github.com/LNP-BP/rust-bitcoin", branch = "staging" }
+bitcoin = { git = "https://github.com/LNP-BP/rust-bitcoin", branch = "staging", features = ["rand"] }
```
Comment (dr-orlovsky) [inline Cargo.toml:35]: We will need this only in test configuration, so can you pls move this feature to `dev-dependencies` it as described in https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#development-dependencies

File: src/bp/strict_encoding.rs [hunk 2]
```diff
@@ -374,3 +374,277 @@ impl StrictDecode for bip32::ExtendedPubKey {
...
+
+        // TODO: Network::Other failing the below assertion
+        //let (decoded, written) = encode_decode(&other).unwrap();
+        //assert_eq!(written, 4);
+        //assert_eq!(decoded, other);
+    }
+
+    #[test]
+    fn test_encoding_pubkey() {
+        static PK_BYTES: [u8; 33] = [
```
Comment (dr-orlovsky) [inline src/bp/strict_encoding.rs:434]: I propose to add test vectors with the following options:
- key starting with 0x02
- ZERO_KEY (must be a constant somewhere underneath `secp256k1`)
- ONE_KEY (idem)

File: src/bp/strict_encoding.rs [hunk 3]
```diff
@@ -374,3 +374,277 @@ impl StrictDecode for bip32::ExtendedPubKey {
...
+
+        let (decoded, written) = encode_decode(&ext_pubkey2).unwrap();
+
+        assert_eq!(written, 79);
+        assert_eq!(ext_pubkey2, decoded);
+    }
+
+    #[test]
+    fn test_encoding_script() {
+        let data = Vec::<u8>::from_hex(
```
Comment (dr-orlovsky) [inline src/bp/strict_encoding.rs:640]: Can we pls test more scripts of different type here?

File: Cargo.toml (hunk 1 above)
Comment (rajarshimaitra) [reply Cargo.toml:35]: Done

File: src/bp/strict_encoding.rs (hunk 2 above)
Comment (rajarshimaitra) [reply src/bp/strict_encoding.rs:434]: ZERO_KEY is removed from secp256k1 library. OKE_KEY test is added.

File: src/bp/strict_encoding.rs (hunk 3 above)
Comment (rajarshimaitra) [reply src/bp/strict_encoding.rs:640]: Added.

**Comment by rajarshimaitra** [comment]
Updated as per the review. This PR is now ready for 2nd look.

**Comment by dr-orlovsky** [comment]
As for garbage, I think PubKey, Signature and OutPoints are enough

**Comment by rajarshimaitra** [comment]
Added garbage tests for `signature`, `pubkey` and `outpiont`. They check for `UnexpectedEof` for now. If that's inadequate and other checks are needed will update accordingly.

**Comment by rajarshimaitra** [comment]
Added `DataIntegrityError` garbage test for pubkey.

**Pull request merged**